packages/telemetry/src/events/session.ts

```diff
@@ -0,0 +1,74 @@
+import escalade from 'escalade/sync';
+import { createRequire } from 'node:module';
+import { fileURLToPath } from 'node:url';
+
+const require = createRequire(import.meta.url);
+
+const EVENT_SESSION = 'ASTRO_CLI_SESSION_STARTED';
+
+interface EventCliSession {
+	astroVersion: string;
+	cliCommand: string;
+}
+
+interface ConfigInfo {
+	hasViteConfig: boolean;
+	hasMarkdownPlugins: boolean;
+	hasBase: boolean;
+	adapter: string | null;
+	integrations: string[];
+	experimentalFeatures: string[];
+}
+
+interface EventCliSessionInternal extends EventCliSession {
+	nodeVersion: string;
+	viteVersion: string;
+	config?: ConfigInfo;
+}
+
+function getViteVersion() {
+	try {
+		const { version } = require('vite/package.json');
+		return version;
+	} catch (e) {}
+	return undefined;
+}
+
+function getExperimentalFeatures(astroConfig?: Record<string, any>): string[] | undefined {
+	if (!astroConfig) return undefined;
+	return Object.entries(astroConfig.experimental).reduce((acc, [key, value]) => {
+		if (value) {
+			acc.push(key);
+		}
+		return acc;
+	}, [] as string[]);
+}
+
+export function eventCliSession(
+	event: EventCliSession,
+	astroConfig?: Record<string, any>
+): { eventName: string; payload: EventCliSessionInternal }[] {
+	const payload: EventCliSessionInternal = {
+		cliCommand: event.cliCommand,
+		// Versions
+		astroVersion: event.astroVersion,
+		viteVersion: getViteVersion(),
+		nodeVersion: process.version.replace(/^v?/, ''),
+		// Config Values
+		config: astroConfig
```

Comment:
cc @FredKSchott this should maybe just be an array of keys?

Reply:
Discussed offline, sending the entire astroConfig could de-anonymous users. Instead we should turn this into an object that contains the features of the config that's used, but not any values.

Reply:
Bah, Nate already did do this.

Reply:
Going to mark this as resolved, no de-anonymizing concerns are present.

Reply:
That nate guy is smart. 

We should definitely use `anonymousConfig` instead of `astroConfig` throughout this part of the codebase, so that it's obvious to other users that this is happening. If it was unclear to us, it could be unclear to others.

Giving `type AnonymousConfig = Record<string, any>` would also help.